mars/tensor/expressions/datasource/from_dataframe.py

```diff
@@ -0,0 +1,21 @@
+from .core import TensorHasInput
+from .... import opcodes as OperandDef
+from ....serialize import KeyField
+from ....dataframe.expressions.utils import build_empty_df
+
+
+class TensorDataFrameDataSource(TensorHasInput):
+    """ represent tensor from DataFrame """
+
+    _op_type_ = OperandDef.TENSOR_FROM_DATAFRAME
+    _input = KeyField('_input')
+
+    def __init__(self, dtype=None, gpu=None, **kw):
+        super(TensorDataFrameDataSource, self).__init__(_dtype=dtype, _gpu=gpu,
+                                                        _sparse=True, **kw)
+
+
+def from_dataframe(in_df):
+    empty_pdf = build_empty_df(in_df.dtypes)
+    op = TensorDataFrameDataSource(dtype=empty_pdf.dtypes[0], gpu=in_df.op.gpu)
```

Comment:
Here I think using `empty_pdf.dtypes[0]` as `dtype` of the tensor op is not correct. The executed result is correct but I think you may see something like

```python
x = md.DataFrame({"x": [1,2,3], "y": [4., 5., 6.]})
y = x.to_tensor()
y.dtype # dtype('int64')
y.executed().dtype # dtype('float64')
```

You may use [find_common_type](https://github.com/pandas-dev/pandas/blob/v0.24.2/pandas/core/dtypes/cast.py#L1072) to find the correct type of result tensor.

Reply:
Thank you. I will open a issue to fix it.

Reply:
Oh, I didn't notice that the dtype is gotten by `empty_pdf.dtypes[0]`, which I suggest is that is creating an empty pandas df, and convert it to numpy to get the right dtype. That is

```python
empty_pdf = build_empty_df(in_df.dtypes)
right_dtype = empty_pdf.to_numpy().dtype
```

This may be also a right way, but `find_common_type` seems more elegant solution anyway. Thank you for bring up this issue @sighingnow .